instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/instrumentation.rb

```diff
@@ -11,6 +11,10 @@ module Instrumentation
     module GraphQL
       # The Instrumentation class contains logic to detect and install the GraphQL instrumentation
       class Instrumentation < OpenTelemetry::Instrumentation::Base
+        compatible do
+          gem_version < Gem::Version.new('3.0.0')
```

Comment:
Don't have overly strong feelings here, but whats the motivation for adding what appears to be a pessimistic guard for future major releases? Did we get burned somewhere by optimistically assuming we'll work with future releases of GraphQL?

Reply:
That is my assertion with any major releases yes.